src/value_generators/src/DocumentGenerator.cpp

```diff
@@ -937,6 +938,53 @@ class ObjectIdGenerator : public Generator<bsoncxx::types::b_oid> {
     const UniqueGenerator<std::string> _hexGen;
 };
 
+/**
+ * `{^UUID: {hex: "3b241101-e2bb-4255-8caf-4136c566a962"} }`
+ *  Generator for non-legacy UUIDs (0x04). hex can itself be a generator so the string can be
+ *  generated in many fashions ^Join, ^FormatString, ... etc.
+ *
+ *  The input format accepted for the hex field is 8-4-4-4-12 format. The code is agnostic as to the
+ *  UUID version.
+ *
+ *  see https://en.wikipedia.org/wiki/Universally_unique_identifier#Version_4_(random)
+ */
+class UuidGenerator : public Generator<bsoncxx::types::b_binary> {
+public:
+    UuidGenerator(const Node& node, GeneratorArgs generatorArgs)
+        : _rng{generatorArgs.rng},
+          _node{node},
+          _hexGen{stringGenerator(node["hex"], generatorArgs)},
+          _subTypeGen{intGenerator(node["subType"],
+                                   generatorArgs,
+                                   static_cast<int64_t>(bsoncxx::binary_sub_type::k_uuid))} {}
+
+    bsoncxx::types::b_binary evaluate() override {
+        hex = _hexGen->evaluate();
+        return bsoncxx::types::b_binary{generateSubType(),
+                                        static_cast<uint32_t>(hex.length()),
```

Comment:
@markbenvenuto  is concerned about the static cast if evaluate is returning a string rather than raw bytes. You may need a call to hex2bytes (which may or may not exist). The std::string can be cast to binary, but may not be the bits you want. 

We looked at your examples and put this together: https://gchq.github.io/CyberChef/#recipe=From_Base64('A-Za-z0-9%2B/%3D',true,false)&input=TTJJeU5ERXhNREV0WlRKaVlpMDBNalUxTFRoallXWXROREV6Tm1NMU5qWmhPVFl5

The output should not be readable with dashes after the conversion.

Reply:
I think we are looking at the wrong line, but it's still wrong. It's the reinterpret_cast on the following line, but I think the issue is the same. Additionally, you have not removed the dashes.